server/commands/collectionImporter.ts

```diff
@@ -1,206 +0,0 @@
-      // there is also a "Name (Imported)" but this is a case not worth dealing
-      // with right now
-      if (!isCreated) {
-        const name = `${item.name} (Imported)`;
-        collection = await Collection.create({
-          teamId: user.teamId,
-          createdById: user.id,
-          name,
-          permission: "read_write",
-        });
-        await Event.create({
-          name: "collections.create",
-          collectionId: collection.id,
-          teamId: collection.teamId,
-          actorId: user.id,
-          data: {
-            name,
-          },
-          ip,
-        });
-      }
-
-      collections[item.path] = collection;
-      continue;
-    }
-
-    if (item.type === "document") {
-      const collectionDir = item.dir.split("/")[0];
-      const collection = collections[collectionDir];
-      invariant(collection, `Collection must exist for document ${item.dir}`);
-
-      // we have a document
-      const content = await item.item.async("string");
-      const name = path.basename(item.name);
-      const tmpDir = os.tmpdir();
-      const tmpFilePath = `${tmpDir}/upload-${uuidv4()}`;
-      await fs.promises.writeFile(tmpFilePath, content);
-
-      const file = new File({
-        name,
-        type: "text/markdown",
-        path: tmpFilePath,
-      });
-      const { text, title } = await documentImporter({
-        file,
-        user,
-        ip,
-      });
-
-      await fs.promises.unlink(tmpFilePath);
-      // must be a nested document, find and reference the parent document
-      let parentDocumentId;
-
-      if (item.depth > 1) {
-        const parentDocument =
-          documents[`${item.dir}.md`] || documents[item.dir];
-        invariant(parentDocument, `Document must exist for parent ${item.dir}`);
-        parentDocumentId = parentDocument.id;
-      }
-
-      const document = await documentCreator({
-        source: "import",
-        title,
-        text,
-        publish: true,
-        collectionId: collection.id,
-        createdAt: item.metadata.createdAt
-          ? new Date(item.metadata.createdAt)
-          : item.item.date,
-        updatedAt: item.item.date,
-        parentDocumentId,
-        user,
-        ip,
-      });
-      documents[item.path] = document;
-      continue;
-    }
-
-    if (item.type === "attachment") {
-      const buffer = await item.item.async("nodebuffer");
-      const attachment = await attachmentCreator({
-        source: "import",
-        name: item.name,
-        type,
-        buffer,
-        user,
-        ip,
-      });
-      attachments[item.path] = attachment;
-      continue;
-    }
-
-    Logger.info("commands", `Skipped importing ${item.path}`);
-  }
-
-  // All collections, documents, and attachments have been created - time to
-  // update the documents to point to newly uploaded attachments where possible
-  for (const attachmentPath of keys(attachments)) {
```

Comment:
This second loop and re-save of documents was removed, the text replacement now happens before the initial save. Something that was enabled by a two-pass approach.